awx/api/views/__init__.py

```diff
@@ -406,13 +409,15 @@ class InstanceInstanceGroupsList(InstanceGroupMembershipMixin, SubListCreateAtta
     def is_valid_relation(self, parent, sub, created=False):
         if parent.node_type == 'control':
             return {'msg': _(f"Cannot change instance group membership of control-only node: {parent.hostname}.")}
+        if parent.node_type == 'hop':
+            return {'msg': _(f"Cannot change instance group membership of hop node: {parent.hostname}.")}
         return None
 
 
 class InstanceHealthCheck(GenericAPIView):
 
     name = _('Instance Health Check')
-    model = models.Instance
+    queryset = models.Instance.objects.exclude(node_type='hop')
```

Comment:
Right, you can't run a health check on a hop node.

It might be good to have a test case that you attempt to visit `/api/v2/instances/2/health_check/` and you get an error code. I don't know what error code it should be, but I want to double-check with you that this is working as intended.

Also, the serializer shouldn't link to this if it's a hop node. I don't see that in this diff

Reply:
tested, and this case gives a 404.

It's consistent behavior, I'm totally fine with the solution here.

Reply:
Excellent.